pkg/gemini/provider.go

```diff
@@ -49,112 +49,150 @@ func (p *Provider) ExtractText(ctx context.Context, config providers.Config, ima
 		mimeType = "image/jpeg"
 	}
 
-	// Prepare request body for Gemini API
-	requestBody := map[string]any{
-		"contents": []map[string]any{
-			{
-				"parts": []map[string]any{
-					{
-						"text": config.Prompt,
-					},
-					{
-						"inline_data": map[string]any{
-							"mime_type": mimeType,
-							"data":      imageBase64,
-						},
-					},
-				},
-			},
-		},
-		"generationConfig": map[string]any{
-			"temperature":     config.Temperature,
-			"mediaResolution": config.MaxResolution,
-		},
+	// Helper to determine next resolution
+	getNextResolution := func(current string) string {
+		switch current {
+		case "", "MEDIA_RESOLUTION_UNSPECIFIED":
+			return "MEDIA_RESOLUTION_HIGH"
+		case "MEDIA_RESOLUTION_HIGH":
+			return "MEDIA_RESOLUTION_MEDIUM"
+		case "MEDIA_RESOLUTION_MEDIUM":
+			return "MEDIA_RESOLUTION_LOW"
+		default:
+			return ""
+		}
 	}
 
+	currentResolution := config.MaxResolution
+	
 	// Use the specified model, default to gemini-pro-vision if not specified
 	model := config.Model
 	if model == "gpt-4o" || model == "" {
 		model = "gemini-pro-vision"
 	}
+	
+	url := fmt.Sprintf("https://generativelanguage.googleapis.com/v1beta/models/%s:generateContent?key=%s", model, apiKey)
 
-	requestJSON, err := json.Marshal(requestBody)
-	if err != nil {
-		return "", providers.UsageInfo{}, fmt.Errorf("failed to marshal request: %w", err)
-	}
+	for {
```

Comment:
Lets make this a little clearer we only every intend to make four API calls at the max
```suggestion
	for i := 0; i < 4; i++ {
```